src/java/org/apache/cassandra/net/async/NettyFactory.java

```diff
@@ -125,10 +126,19 @@
      * Determine the number of accept threads we need, which is based upon the number of listening sockets we will have.
      * We'll have either 1 or 2 listen sockets, depending on if we use SSL or not in combination with non-SSL. If we have both,
      * we'll have two sockets, and thus need two threads; else one socket and one thread.
+     *
+     * If the operator has configured multiple IP addresses (both {@link org.apache.cassandra.config.Config#broadcast_address}
+     * and {@link org.apache.cassandra.config.Config#listen_address} are configured), then we listen on another set of sockets
+     * - basically doubling the count. See CASSANDRA-9748 for more details.
      */
     static int determineAcceptGroupSize(InternodeEncryption internode_encryption)
     {
-        return internode_encryption == InternodeEncryption.dc || internode_encryption == InternodeEncryption.rack ? 2 : 1;
+        int listenSocketCount = internode_encryption == InternodeEncryption.dc || internode_encryption == InternodeEncryption.rack ? 2 : 1;
+
+        if (MessagingService.shouldListenOnBroadcastAddress())
+            listenSocketCount *= 2;
```

Comment:
Why do you need more threads to listen on more sockets? It's a non-blocking operation and more sockets doesn't indicate a need for more throughput?